meson.build

```diff
@@ -165,25 +165,47 @@ endif
 # Dependencies
 dl_dep = cc.find_library('dl', required: false)
 gl_dep = dependency('gl', required: false)
-egl_dep = dependency('egl', required: false)
-elg_headers_dep = egl_dep.partial_dependency(compile_args: true, includes: true)
 
 # Optional dependencies for tests
 x11_dep = dependency('x11', required: false)
 x11_headers_dep = x11_dep.partial_dependency(compile_args: true, includes: true)
 
-# GLES v2 and v1 may have pkg-config files, courtesy of downstream
-# packagers; let's check those first, and fall back to find_library()
-# if we fail
+# We have multiple checks for EGL and GLES v2/v1 to support different providers:
+# 1. pkg-config for Mesa
+# 2. find_library() for ANGLE, which do not support pkg-config nor CMake.
+#    Note that Microsoft's "link" requires "lib" prefix.
+# 3. CMake for Qt 5, which bundles ANGLE.
+egl_dep = dependency('egl', required: false)
+if not egl_dep.found()
+  egl_dep = cc.find_library('EGL', required: false)
+endif
+if not egl_dep.found() and host_machine == 'windows'
```

Comment:
Testing on darwin with meson `1.4.0`:
```
../meson.build:182:43: ERROR: 
Trying to compare values of different types (MachineInfo, str) using ==.
This was deprecated and undefined behavior previously and is as of 0.60.0 a hard error.
```

I suppose this can be fixed with `host_machine.system() == 'windows'` here, and the other two lines below?

Reply:
Actually, you can use `host_system` already defined above :)